internal/build/fakes/fake_phase_manager.go

```diff
@@ -0,0 +1,104 @@
+package fakes
+
+import "github.com/buildpacks/pack/internal/build"
+
+type FakePhaseManager struct {
```

Comment:
This looks like a lot of boilerplate. Is this auto-generated? If not, I wouldn't look forward to maintaining this...

Reply:
It's not autogenerated, which is a purposeful choice to reduce our set of dependencies. An auto-generated version of this file would be many multiples in length as it would be required to solve for the generic problem of multiple types of response & assertions, whereas with this hand-rolled fake we can optimize for the expected use cases (with guards against unexpected use)

Reply:
> reduce our set of dependencies

what type of dependencies are referring to ?

> auto-generated version of this file would be many multiples in length

But given that it would be auto-generated, do we care?

> we can optimize for the expected use cases

Not entire sure what optimize here means. If it's size or performance, it's not really applicable since it's only used within the testing scope. If it's usage, I'm not convinced that using something else wouldn't allow you to yield at the same results.

Reply:
We could use something like [counterfeiter](https://github.com/maxbrunsfeld/counterfeiter) to generate these fakes. Would this be acceptable?

Reply:
With the latest changes, the fakes are a lot simpler and shouldn't need to change as often.